.github/workflows/e2e.yaml

```diff
@@ -0,0 +1,63 @@
+name: e2e-tests
+
+
+on: pull_request
+
+env:
+  # Make sure to exit early if cache segment download times out after 2 minutes.
+  # We limit cache download as a whole to 5 minutes.
+  SEGMENT_DOWNLOAD_TIMEOUT_MINS: 2
+jobs:
+  kubernetes-test:
+    env:
+      KUBECONFIG: /tmp/kube-config-splunk-otel-collector-chart-e2e-testing
+    strategy:
+      fail-fast: false
+      matrix:
+        k8s-version:
+          - v1.28.0 # Support: 28 Aug 2024 - 28 Oct 2024
+          - v1.27.3  # Support: 28 Apr 2024 - 28 Jun 2024
+          - v1.26.6 # Support: 28 Dec 2023 - 28 Feb 2024
+          - v1.25.11 # Support: 27 Aug 2023 - 27 Oct 2023
+          - v1.24.15  # Support: 28 May 2023 - 28 Jul 2023
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v4
+        with:
+          go-version: ~1.20.8
+          cache: false
+      - name: Cache Go
+        id: go-cache
+        timeout-minutes: 5
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~/go/bin
+            ~/go/pkg/mod
+          key: go-cache-${{ runner.os }}-${{ hashFiles('**/go.sum') }}
+      - name: Create kind cluster
+        uses: helm/kind-action@v1.8.0
```

Comment:
Requiring the kind cluster to exist before the tests (and be managed by a gh action) makes these less practical for local development and debugging. Since you're running a single control plane node there's really no reason to imo. The splunk collector distro demonstrates how to do this w/o coupling to gh.

Reply:
I copied the approach taken by upstream with e2e tests here: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/.github/workflows/e2e-tests.yml#L73

Definitely can do more there.

Reply:
Using a make action can be useful here for gh workflows and local environments. I added this ['make start-kind](https://github.com/signalfx/splunk-otel-collector-operator/blob/e801980387bac9654c4cfb241f05b0fb5d0f3373/Makefile#L210)' action as a possible solution addressing pain points similar to this.